mypy/plugins/ctypes.py

```diff
@@ -116,19 +117,27 @@ def array_constructor_callback(ctx: 'mypy.plugin.FunctionContext') -> Type:
         allowed = _autoconvertible_to_cdata(et, ctx.api)
         assert len(ctx.arg_types) == 1, \
             "The stub of the ctypes.Array constructor should have a single vararg parameter"
-        for arg_num, arg_type in enumerate(ctx.arg_types[0], 1):
-            # TODO This causes false errors if the argument list contains *args.
-            # In a function hook, the type of an *args parameter is the type of the iterable being
-            # unpacked. However, FunctionContext currently doesn't provide a way to differentiate
-            # between normal arguments and *args, so the iterable type is considered invalid.
-            # Once FunctionContext has an API for this, *args should be allowed here if the
-            # iterable's element type is compatible with the array element type.
-            if not is_subtype(arg_type, allowed):
+        for arg_num, (arg_kind, arg_type) in enumerate(zip(ctx.arg_kinds[0], ctx.arg_types[0]), 1):
+            if arg_kind not in {nodes.ARG_POS, nodes.ARG_STAR}:
+                ctx.api.msg.fail(
+                    "Array constructor does not allow keyword arguments",
```

Comment:
Please add a test that triggers this error message.

Reply:
It looks like this actually isn't ever called -- it looks like the constructor in typeshed `def __init__(self, *args: Any) -> None)` definition takes care of this case.

Reply:
Oh, wait, so if it is never going to be called, maybe we can remove this code?

Reply:
Done!